### ISSUE TITLE
Are there any way to be notified when the SASKey is regenerated

### ISSUE THREAD

**Issue by hbiarge** [Azure/azure-service-bus#81]
# Description
When an app is running and listening for messages in a topic suscription, if I regenerate the SASKey that the app uses to connect to the suscription, and don't shut down the app, the app remains receiving messages.

Are there any way to be notified when the SASKey changes, in order to be able to renew the connection to the suscription or stop the reception of messages?

If there is not any way in the client to be notified about this, what is the best strategy to acomplish this scenario?

Thank you!

**Comment by SeanFeldman** [comment]
I'm tempted to say EventGrid _could_ do that in the future, but that would be a wild guess on my part.

**Comment by binzywu** [comment]
@SeanFeldman, you are right. We will be starting publishing some events to a system Topic in EventGrid in near future. This is a good candidate for the events.

**Comment by hbiarge** [comment]
So I understand that this feature will be available as a Service Event in the integration of Azure Service Bus with EventGrid and it won't be available in the Azure Service Bus SDK, am I right?

**Comment by EldertGrootenboer** [comment]
We have opened an investigation task for this in our backlog, and will update this issue when we have more information.

**Comment by EldertGrootenboer** [comment]
Thank you for your feedback on this item, it helps us in our efforts to continuously prioritize the different requests we get from all our various feedback channels. We have added this feature in our backlog, however we currently don't have an ETA on when development might start on this. For now, to help us give this the right priority, it would be helpful to see others vote and support this feature, as well as explain their scenarios.

**Comment by EldertGrootenboer** [comment]
This feature in our backlog, however we currently don't have an ETA on when development might start on this. For now, to help us give this the right priority, it would be helpful to see others vote and support this feature, as well as explain their scenarios.
